### ISSUE TITLE
nf-core download skips 'bowtie2 align' singularity image with container string parsing error

### ISSUE THREAD

**Issue by rdwngs** [nf-core/chipseq#318]
### Description of the bug

When attempting to download and install chipseq-2.0.0 with singularity containers, the 'bowtie2 align' image gives a container string parsing error. I noticed the container string in main.nf for this module has nested double quotes instead of internal single quotes - is it terminating early?

container "${ workflow.containerEngine == "singularity" && !task.ext.singularity_pull_docker_container ?
        "https://depot.galaxyproject.org/singularity/mulled-v2-ac74a7f02cebcfcc07d8e8d1d750af9c83b4d45a:1744f68fe955578c63054b55309e05b41c37a80d-0" :
        "quay.io/biocontainers/mulled-v2-ac74a7f02cebcfcc07d8e8d1d750af9c83b4d45a:1744f68fe955578c63054b55309e05b41c37a80d-0" }"

### Command used and terminal output

```console
nf-core download chipseq -c singularity

...
ERROR    Cannot parse container string in                                                                download.py:483
         'nf-core-chipseq-2.0.0/workflow/modules/nf-core/modules/bowtie2/align/main.nf':

             ${ workflow.containerEngine ==

         ⚠ Skipping this singularity image..
```

### Relevant files

https://github.com/nf-core/chipseq/blob/master/modules/nf-core/modules/bowtie2/align/main.nf

### System information

nextflow version 22.10.0.5826
nf-core version 2.6
HPC
Local executor
Singularity container engine
Redhat Linux 8.5
chipseq version 2.0.0

**Comment by isaacvock** [comment]
I am having the exact same problem with chipseq version 2.0.0 and nf-core download, and was wondering if there were any updates; very similar system information as in the original report, but here are my details:

nextflow version 22.10.4.5836
nf-core version 2.7.2
HPC
Local executor
Redhat Linux 7.9

**Comment by isaacvock** [comment]
Digging a little deeper, looks like most recent change to the main.nf file that @rdwngs noted saw the single quotes changed to double quotes, and subsequent failure of CI tests. Therefore, I am going to make a pull request to revert this change that is likely causing the problem.

**Comment by MatthiasZepper** [comment]
Sadly, downloading exactly this pipeline version that contains the affected module will not be possible to support with the current nf-core download, [for reasons explained here](https://github.com/nf-core/tools/issues/2392#issuecomment-1682393518). 

Nextflow has received a new functionality `nextflow inspect` that does resolve the container correctly, but it will take a while until the whole codebase of `nf-core download` has been rewritten to use the outputs of inspect (Certainly not for version 2.10, possibly also not yet 2.11). Thus, I suppose, the next pipeline release will happen earlier.

**Comment by JoseEspinosa** [comment]
Fixed in #339
